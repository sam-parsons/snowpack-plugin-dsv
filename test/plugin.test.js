const plugin = require('../index.js');

describe('snowpack dsv plugin', () => {
  test('invokes plugin without error', () => {
    expect(plugin()).toHaveProperty('name');
  });

  test('load function return value matches snapshot with csv input', () => {
    const obj = plugin();
    const res = obj.load({
      fileExt: '.csv',
      filePath: './test/fixtures/data.csv',
    });
    expect(res).toMatchSnapshot();
  });

  test('load function return value matches snapshot with tsv input', () => {
    const obj = plugin();
    const res = obj.load({
      fileExt: '.tsv',
      filePath: './test/fixtures/data.tsv',
    });
    expect(res).toMatchSnapshot();
  });
});
