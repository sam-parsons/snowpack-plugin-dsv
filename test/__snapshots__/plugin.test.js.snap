// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snowpack dsv plugin load function return value matches snapshot with csv input 1`] = `
"export default [ { type:\\"furry\\",
    \\" value;\\":\\" true;\\" },
  { type:\\"fuzzy\\",
    \\" value;\\":\\" false;\\" } ];"
`;

exports[`snowpack dsv plugin load function return value matches snapshot with tsv input 1`] = `
"export default [ { powerful:\\"stand\\",
    drink:\\"orange\\",
    would:\\"slightly\\",
    say:\\"none\\",
    parallel:\\"tongue\\",
    note:\\"sudden\\" },
  { powerful:\\"solve\\",
    drink:\\"aid\\",
    would:\\"ourselves\\",
    say:\\"dear\\",
    parallel:\\"from\\",
    note:\\"corn\\" },
  { powerful:\\"beneath\\",
    drink:\\"hardly\\",
    would:\\"count\\",
    say:\\"yesterday\\",
    parallel:\\"broad\\",
    note:\\"hair\\" },
  { powerful:\\"should\\",
    drink:\\"sing\\",
    would:\\"rocket\\",
    say:\\"mice\\",
    parallel:\\"scale\\",
    note:\\"page\\" },
  { powerful:\\"disappear\\",
    drink:\\"additional\\",
    would:\\"cell\\",
    say:\\"steam\\",
    parallel:\\"map\\",
    note:\\"second\\" } ];"
`;
